app/src/main/java/com/infomaniak/mail/utils/LocalStorageUtils.kt

```diff
@@ -87,21 +90,59 @@ object LocalStorageUtils {
         return File(generateRootDir(context.attachmentsUploadRootDir, userId, mailboxId), draftLocalUuid)
     }
 
-    fun saveUploadAttachment(context: Context, uri: Uri, fileName: String, draftLocalUuid: String): File? {
+    fun saveUploadAttachment(
+        context: Context,
+        uri: Uri,
+        fileName: String,
+        draftLocalUuid: String,
+        snackbarManager: SnackbarManager,
+    ): File? {
         return context.contentResolver.openInputStream(uri)?.use { inputStream ->
             val attachmentsUploadDir = getAttachmentsUploadDir(context, draftLocalUuid)
             if (!attachmentsUploadDir.exists()) attachmentsUploadDir.mkdirs()
+
             val hashedFileName = "${uri.toString().substringAfter("document/").hashCode()}_$fileName"
-            File(attachmentsUploadDir, hashedFileName).also { file ->
-                FileOutputStream(file).use(inputStream::copyTo)
-            }
+
+            return@use getFileToUpload(context, uri, snackbarManager, attachmentsUploadDir, hashedFileName, inputStream)
         } ?: run {
+            Sentry.captureMessage("failed to access uri")
+
+            null
+        }
+    }
+
+    private fun getFileToUpload(
+        context: Context,
+        uri: Uri,
+        snackbarManager: SnackbarManager,
+        attachmentsUploadDir: File,
+        hashedFileName: String,
+        inputStream: InputStream,
+    ): File? {
+        val file = File(attachmentsUploadDir, hashedFileName)
+        val isSuccess = runCatching {
+            FileOutputStream(file).use(inputStream::copyTo)
+            true
+        }.getOrElse {
             Sentry.withScope { scope ->
-                scope.setExtra("uri", uri.toString())
-                Sentry.captureMessage("failed to access uri")
+                scope.setExtra("uri", uri.toString().replace(file.path, "HIDDEN FILE NAME"))
+                val exception =
+                    it.message?.let { AttachmentMissingFileException(it.replace(file.path, "HIDDEN FILE NAME")) }
+                        ?: it
```

Comment:
There's 2 different `it` here, rename at least one of them _(or maybe all of them)_.